### PR TITLE
Delete Kafka consumer group when KafkaSource is deleted

### LINKED ISSUE #2087: Resource clean up of auto generated consumer groups
**Problem**
Consumer groups are created if not specified when a new Kafka source is created; however if the source is later deleted this automatically consumer group is not removed.

Expected behavior
Removing kafasource that had no consumer group specified should remove the auto generated consumer group

References: https://github.com/knative-sandbox/eventing-kafka/issues/839

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?

**Exit Criteria**
A measurable (binary) test that would indicate that the problem has been resolved.

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?

**Additional context (optional)**
Add any other context about the feature request here.

/area source
/area data-plane

### PR DESCRIPTION
**Pull request by aavarghese**
Signed-off-by: aavarghese <avarghese@us.ibm.com>

Fixes #2087 

## Proposed Changes

- Delete Kafka consumer group when KafkaSource is deleted for resource clean-up

**Release Note**

**Docs**

### CODE REVIEW

**Review by pierDipi — COMMENTED** [review]
Is it possible to do this in the `ConsumerGroup` reconciler?

**Comment by aavarghese** [comment]
> Is it possible to do this in the `ConsumerGroup` reconciler?

Hey @pierDipi any particular reason to do this in CG reconciler? Since we do the create + update of CG in KafkaSource reconciler, seemed ideal to add Delete CG in the same rec. And not along with consumer creation/deletion in CG reconciler. We get all the needed auth info, CG name and bootstrapservers directly from the Kafkasource CR, as well. Wdyt?

File: control-plane/pkg/reconciler/consumergroup/consumergroup.go [hunk 1]
```diff
@@ -105,6 +110,48 @@ func (r Reconciler) ReconcileKind(ctx context.Context, cg *kafkainternals.Consum
 	return nil
 }
 
+func (r *Reconciler) FinalizeKind(ctx context.Context, cg *kafkainternals.ConsumerGroup) reconciler.Event {
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/consumergroup/consumergroup.go:113]: With this new `FinalizeKind` method, I was expecting to see that tests in `consumergroup_test.go` for `ReconcileKind` (TestReconcileKind) would have a new finalizer update event emitted.

Comment (aavarghese) [reply control-plane/pkg/reconciler/consumergroup/consumergroup.go]: Thank you!!!

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Comment by aavarghese** [comment]
```
java.lang.IllegalStateException: Potential control-plane bug: failed to get verticle
	at dev.knative.eventing.kafka.broker.dispatcher.main.ConsumerDeployerVerticle.onNewEgress(ConsumerDeployerVerticle.java:110)
	at dev.knative.eventing.kafka.broker.core.reconciler.impl.ResourcesReconcilerImpl.lambda$reconcileEgress$10(ResourcesReconcilerImpl.java:110)
```

**Pull request merged**